File: check_km.mjs
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY

const supabase = createClient(supabaseUrl, supabaseKey)

async function run() {
  const { data: vehicles } = await supabase.from('vehicles').select('id, current_km, license_plate')
  for (const v of vehicles) {
    const { data: bookings } = await supabase.from('bookings').select('starting_km, return_km').eq('vehicle_id', v.id)
    const { data: handovers } = await supabase.from('vehicle_handovers').select('pickup_km, return_km').eq('vehicle_id', v.id)
    
    let maxKm = 0
    if (bookings) {
      for (const b of bookings) {
        if (b.starting_km && b.starting_km > maxKm) maxKm = b.starting_km
        if (b.return_km && b.return_km > maxKm) maxKm = b.return_km
      }
    }
    if (handovers) {
      for (const h of handovers) {
        if (h.pickup_km && h.pickup_km > maxKm) maxKm = h.pickup_km
        if (h.return_km && h.return_km > maxKm) maxKm = h.return_km
      }
    }
    
    if (maxKm > (v.current_km || 0)) {
      console.log(`-> Fixing ${v.license_plate} to ${maxKm}`)
      const res = await supabase.from('vehicles').update({ current_km: maxKm }).eq('id', v.id)
      if (res.error) console.error("Error for", v.license_plate, ":", res.error)
      else console.log("Success for", v.license_plate)
    }
  }
}
run().catch(console.error)

File: check_km.py
import asyncio
from supabase import create_client
import os
from dotenv import load_dotenv

load_dotenv(r"c:\Users\asus\Desktop\projet\car-dash\.env.local")
supabase_url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
if not supabase_key:
    supabase_key = os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY")
supabase = create_client(supabase_url, supabase_key)

res = supabase.table("vehicles").select("id, current_km, license_plate").execute()
for v in res.data:
    b_res = supabase.table("bookings").select("starting_km, return_km").eq("vehicle_id", v["id"]).execute()
    h_res = supabase.table("vehicle_handovers").select("pickup_km, return_km").eq("vehicle_id", v["id"]).execute()
    
    max_km = 0
    for b in b_res.data:
        if b.get("starting_km") and b["starting_km"] > max_km: max_km = b["starting_km"]
        if b.get("return_km") and b["return_km"] > max_km: max_km = b["return_km"]
    for h in h_res.data:
        if h.get("pickup_km") and h["pickup_km"] > max_km: max_km = h["pickup_km"]
        if h.get("return_km") and h["return_km"] > max_km: max_km = h["return_km"]
        
    print(f"Vehicle {v['license_plate']} Max KM: {max_km} (Current DB: {v['current_km']})")
    
    if max_km > (v["current_km"] or 0):
        print(f"-> Fixing {v['license_plate']} to {max_km}")
        supabase.table("vehicles").update({"current_km": max_km, "current_mileage": max_km}).eq("id", v["id"]).execute()
